go/vt/vtgate/tabletgateway.go

```diff
@@ -57,7 +59,7 @@ var (
 func init() {
 	servenv.OnParseFor("vtgate", func(fs *pflag.FlagSet) {
 		fs.StringVar(&CellsToWatch, "cells_to_watch", "", "comma-separated list of cells for watching tablets")
-		fs.StringVar(&bufferImplementation, "buffer_implementation", "keyspace_events", "Allowed values: healthcheck (legacy implementation), keyspace_events (default)")
+		fs.StringVar(&bufferImplementationDeprecated, "buffer_implementation", "deprecated", "DEPRECATED: will be deleted in Vitess 19")
```

Comment:
The way to deprecate a flag is to add a line like
```
	fs.MarkDeprecated("throttle_threshold", "Replication lag threshold for default lag throttling")
```
Then you can delete the var completely, and the flag disappears from the help text. See #13246 for example

Reply:
Well, I pointed you to the wrong example. It has to be something like this
```
	fs.Bool("use_super_read_only", true, "Set super_read_only flag when performing planned failover.")
	fs.MarkDeprecated("use_super_read_only", "From v17 onwards MySQL server will always try to start with super_read_only=ON")
```
So, you can't delete the original line, you just add the new one.
Incidentally, it was done incorrectly for throttler flags in #13246 and needs to be fixed.
This is why tests are failing. vtgate doesn't start because it is being provided the `--buffer_implementation` flag and the flag has been completely removed.

Reply:
Thanks, hope I finally got it right ;-) 

I haven't removed the flag from the buffering tests since they helped point out the incorrect implementation. We will remove it in v19 when we will remove the flag entirely.

Reply:
That makes perfect sense. That is in fact the only way to make sure we don't break flags on deprecation.